src/xenia/base/system_gnulinux.cc

```diff
@@ -29,7 +33,19 @@ void LaunchWebBrowser(const std::string_view url) {
   system(cmd.c_str());
 }
 
-void LaunchFileExplorer(const std::filesystem::path& path) { assert_always(); }
+void LaunchFileExplorer(const std::filesystem::path& path) {
+  pid_t pid;
+  int n = path.string().length();
+  char cmdxdg[] = "xdg-open";
+  char cmdpath[n + 1];
+  char* argv[] = {cmdxdg, cmdpath, NULL};
+  int status = -1;
+  strcpy(cmdpath, path.c_str());
+  status = posix_spawn(&pid, "/usr/bin/xdg-open", NULL, NULL, argv, environ);
+  if (status != 0) {
+    perror("posix_spawn");
```

Comment:
Except for the part of the platform-specific initialization where logging still hasn't yet been initialized, we use the Xenia logging (`XELOGE` from `"xenia/base/logging.h"`) for error messages, not `perror`.

Reply:
Right, I was using `perror` as that outputs a description of the error, Not just the error number that is assigned to `status`. Will switch to XELOGE and have a look about getting the error description through that as well.